scenarios/test_codegen/test/lib_tests/EntityHistory_test.res

```diff
@@ -0,0 +1,222 @@
+          -- Find the most recent record for the same id
+          SELECT entity_history_block_timestamp, entity_history_chain_id, entity_history_block_number, entity_history_log_index INTO v_previous_record
+          FROM "public"."TestEntity_history"
+          WHERE id = history_row.id
+          ORDER BY entity_history_block_timestamp DESC, entity_history_chain_id DESC, entity_history_block_number DESC, entity_history_log_index DESC
+          LIMIT 1;
+
+          -- If a previous record exists, use its values
+          IF FOUND THEN
+            history_row.previous_entity_history_block_timestamp := v_previous_record.entity_history_block_timestamp; history_row.previous_entity_history_chain_id := v_previous_record.entity_history_chain_id; history_row.previous_entity_history_block_number := v_previous_record.entity_history_block_number; history_row.previous_entity_history_log_index := v_previous_record.entity_history_log_index;
+            ElSE
+            -- Check if a value for the id exists in the origin table and if so, insert a history row for it.
+            SELECT "id", "fieldA", "fieldB" FROM "public"."TestEntity" WHERE id = history_row.id INTO v_origin_record;
+            IF FOUND THEN
+              INSERT INTO "public"."TestEntity_history" (entity_history_block_timestamp, entity_history_chain_id, entity_history_block_number, entity_history_log_index, "id", "fieldA", "fieldB")
+              -- SET the current change data fields to 0 since we don't know what they were
+              -- and it doesn't matter provided they are less than any new values
+              VALUES (0, 0, 0, 0, v_origin_record."id", v_origin_record."fieldA", v_origin_record."fieldB");
+
+              history_row.previous_entity_history_block_timestamp := 0; history_row.previous_entity_history_chain_id := 0; history_row.previous_entity_history_block_number := 0; history_row.previous_entity_history_log_index := 0;
+            END IF;
+          END IF;
+        END IF;
+
+        INSERT INTO "public"."TestEntity_history" (entity_history_block_timestamp, entity_history_chain_id, entity_history_block_number, entity_history_log_index, previous_entity_history_block_timestamp, previous_entity_history_chain_id, previous_entity_history_block_number, previous_entity_history_log_index, "id", "fieldA", "fieldB")
+        VALUES (history_row.entity_history_block_timestamp, history_row.entity_history_chain_id, history_row.entity_history_block_number, history_row.entity_history_log_index, history_row.previous_entity_history_block_timestamp, history_row.previous_entity_history_chain_id, history_row.previous_entity_history_block_number, history_row.previous_entity_history_log_index, history_row."id", history_row."fieldA", history_row."fieldB");
+      END;
+      $$ LANGUAGE plpgsql;
+      `
+
+    Assert.equal(expected, mockEntityHistory.createInsertFnQuery)
+  })
+
+  Async.it("Creating tables and functions works", async () => {
```

Comment:
This one demonstrates the function and how it will be called. Although each entity will have an actual function it can call in the next PR